src/com/owncloud/android/ui/fragment/ShareFileFragment.java

```diff
@@ -655,6 +740,25 @@ public void onClick(View v) {
                     mOnPasswordInteractionListener
             );
 
+            /// update state of the edit permission switch
+            Switch editPermissionSwitch = getEditPermissionSwitch();
+            if(getEditPermissionSection().getVisibility() == View.VISIBLE) {
```

Comment:
Is this check really needed? Is there a way that the switch is clicked while being invisible?